src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -68,6 +70,14 @@ class AuthScreens extends React.Component {
             authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
         }).then(subscribeToReportCommentEvents);
 
+        // If we have a validation code in our URL ensure we validate the associated login
+        if (this.props.currentURL.includes(`/${ROUTES.VALIDATE_LOGIN}/`)) {
+            // Split the /v/accountID/validateCode URL string into distinct components
+            // "", "v", "accountID", "validateCode"
+            const urlArray = this.props.currentURL.split('/');
+            User.validateLogin(Number(urlArray[2]), urlArray[3]);
```

Comment:
I'm a bit concerned that `validateLogin()` is being called from `componentDidMount()`. That means that `AuthScreens` will render once, then validate the login, then redirect to the settings page if successful. I think what would be better to do is to prevent any rendering of `AuthScreens` at all until the validation API call is done. This could be done using a state variable and an Onyx key (maybe something in the session).